### PR TITLE
PBM-1339 Extended PBM compatibility table

### PR DESCRIPTION
**Pull request by nastena1606**
Added links to it

modified:   docs/details/versions.md
	modified:   docs/index.md
	modified:   docs/install/repos.md
	modified:   docs/install/source.md
	modified:   docs/install/tarball.md
	modified:   docs/installation.md
	modified:   mkdocs-base.yml

Added:
- warning to mention to use old PBM versions with EOL-ed MongoDB versions (even if latest PBM version could still work in this case)

### CODE REVIEW

**Comment by sleto-it** [comment]
Hi @nastena1606, thanks for working on this, and for linking the compatibility matrix in multiple points of the doc.
From my point of view (user perspective) the table is clear - I have clear view of which MongoDB versions I can use with specific PBM versions.

It might need validation from Engineering, I am validating only the user interface here.

One comment, I would suggest bullet points inside the table, for better clarity. I am pushing this change, but feel free to revert if you do not like it. I am reporting below how it looks like after a local compilation:

**Before:**

before

**Now:**

now

**Review by sleto-it — APPROVED** [review]
Pushed a commit to propose using bullet points inside the table for better clarify. OK from my side, from a user perspective. Might need additional validation from Engineering wrt checking the specific versions - otherwise OK to merge. Thanks

**Comment by nastena1606** [comment]
Awesome, thank you @sleto-it !

File: docs/details/versions.md [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+* MongoDB Community / Enterprise Edition  with [MongoDB Replication :octicons-link-external-16:](https://docs.mongodb.com/manual/replication/) enabled for logical backups.
+* Percona Server for MongoDB with [MongoDB Replication :octicons-link-external-16:](https://docs.mongodb.com/manual/replication/) for logical backups. For physical backups, also configure WiredTiger as the storage engine.
 
+The following table provides information about the supported MongoDB versions for each PBM release:
+
+| PBM version | MongoDB Community / Enterprise | Percona Server for MongoDB|Compatibility with previous PBM versions|
+| ----------- |------------------------------- | ------------------------- |----------------------------------------|
+| [2.5.0](../release-notes/2.5.0.md) | <ul><li>Logical backups: version 4.4 and higher</li></ul> | <ul><li>Logical backups: version 4.4 and higher</li><li>Physical backups: version 4.4.6-8 and higher, 5.0.x, 6.0.x, 7.0.x</li><li>Incremental backups: version 4.4.18, 5.0.14-12, 6.0.3-2 and higher, 7.0.x</li></ul> | Yes |
```
Comment (boris-ilijic) [inline docs/details/versions.md:12]: When we made this pbm release (2.5), Mongo 4.4 was already EOL. Do we still want to mention it as supported MongoDB version?
From technical perspective it is still compatible.

Comment (sleto-it) [reply docs/details/versions.md:12]: Good point. If it was already EOL and we did not test this PBM release with it, probably good to remove it, and people still using old MongoDB versions, can consider using an older PBM version to be fully sure it works.  We can consider a general statement in the page like "Old or EOL-ed MongoDB versions not included in the row for a specific PBM release might still work but are not fully certified.  Please use an older PBM version that is certified for your MongoDB version, in this case. " - or something similar, wanted just to give an idea. In the hope it helps,

File: docs/details/versions.md [hunk 2]
```diff
@@ -1,10 +1,27 @@
...
+
+!!! note ""
+
+    End-of-life MongoDB versions may work with PBM, but they are not tested for compatibility. Consider using an previous PBM version in this case.
+
+
+| PBM version | MongoDB Community / Enterprise | Percona Server for MongoDB|Compatibility with previous PBM versions|
+| ----------- |------------------------------- | ------------------------- |----------------------------------------|
+| [2.5.0](../release-notes/2.5.0.md) | <ul><li>Logical backups: version 5.0.x and higher</li></ul> | <ul><li>Logical backups: version 5.0.x and higher</li><li>Physical backups: version 5.0.x, 6.0.x, 7.0.x</li><li>Incremental backups: version 5.0.14-12, 6.0.3-2 and higher, 7.0.x</li></ul> | Yes |
+| [2.4.0](../release-notes/2.4.0.md) | <ul><li>Logical backups: version 4.4 and higher</li></ul> | <ul><li>Logical backups: version 4.4 and higher</li><li>Physical backups - version 4.4.6-8 and higher, 5.0.x, 6.0.x, 7.0.x</li><li>Incremental backups: version 4.4.18, 5.0.14-12, 6.0.3-2 and higher, 7.0.x</li></ul>| Yes |
```
Comment (sleto-it) [inline docs/details/versions.md:18]:
```suggestion
| [2.4.0](../release-notes/2.4.0.md) | <ul><li>Logical backups: version 5.0.x and higher</li></ul> | <ul><li>Logical backups: version 5.0.x and higher</li><li>Physical backups: version 5.0.x, 6.0.x, 7.0.x</li><li>Incremental backups: version 5.0.14-12, 6.0.3-2 and higher, 7.0.x</li></ul>| Yes |
```

Comment (sleto-it) [reply docs/details/versions.md:18]: @nastena1606 did the same for 2.4, as MongoDB 4.4 was already EOL when 2.4 was released (feb. 2024 vs march 2024).  As soon this is implemented, hopefully engineering can review again.  But this way there is a consistent approach (removing eol versions)

File: docs/details/versions.md (hunk 1 above)
Comment (sleto-it) [reply docs/details/versions.md]: A warning message was added at the top of the page "End-of-life MongoDB versions may work with PBM, but they are not tested for compatibility. Consider using an previous PBM version in this case." , and MongoDB versions already EOL when a specific PBM version was released, removed from the corresponding row.

Resolving.

**Review by defbin — APPROVED** [review]

**Pull request merged**